### PR TITLE
add multipolygon

### PR DESCRIPTION
**Pull request by ronaldotantra**
support covering type MultiPolygon

### CODE REVIEW

File: internal/app/controllers/geometry.go [hunk 1]
```diff
@@ -31,6 +31,18 @@ func (u GeometryController) Cover(c *gin.Context) {
 	var s2cells [][][]float64
 
 	for _, f := range fs {
+		if f.Geometry.IsMultiPolygon() {
+			for _, mp := range f.Geometry.MultiPolygon {
+				for _, p := range mp {
```
Comment (pantrif) [inline internal/app/controllers/geometry.go]: It might be better to export this inner iteration to its own private function to avoid duplication

**Review by pantrif — APPROVED** [review]
👍

File: internal/app/controllers/geometry.go (hunk 1 above)
Comment (ronaldotantra) [reply internal/app/controllers/geometry.go]: refactored thanks

**Review by pantrif — APPROVED** [review]

**Comment by ronaldotantra** [comment]
would you mind to merge it ? @pantrif big thanks

**Comment by missinglink** [comment]
Hi @ronaldotantra, I recently completed a pure JS port of s2 and added a supplementary module for working with geojson data.

One of the issues I found with multi* geometries is that the resulting union doesn't satisfy the RegionCoverer constraints such as `maxCells`, `minLevel` etc.

An elegant workaround is to union each member geometry and then pass the union into the RegionCoverer (as `CellUnion` satisfies the `Region` interface) so the constraints are applied.

https://github.com/missinglink/s2js?tab=readme-ov-file#geojson-support

**Pull request merged**